### PR TITLE
feat: [PIE-5083]: update user group input to only list allowed values…

### PR DESCRIPTION
**Pull request by rameshx**
… in templatized view

### Summary

Updates user group input related components/hooks to support filtering of user groups based on passed identifiers.

#### Screenshots

https://user-images.githubusercontent.com/52284933/187210228-35d7c9ed-fe4e-4aba-851a-b20d9e8e052d.mov

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Failed_** 
 


            
 #### 4 Issues 
* Code issue 4 Code issues 

 [Details](https://app.embold.io/organization/gh/harness/repositories/e864badd0b95ab81c060b15b85f15701/pull-request/10873)

**Comment by rameshx** [comment]
retrigger titlecheck

File: src/modules/10-common/components/UserGroupsInput/__test__/__snapshots__/UserGroupsInput.test.tsx.snap [hunk 1]
```diff
@@ -224,29 +224,13 @@ exports[`Rendering Modal should close on clicking cross 1`] = `
...
-                  d="M18 2.084a15.915 15.915 0 010 31.831 15.915 15.915 0 010-31.83"
-                  stroke="currentColor"
-                  stroke-dasharray="75,100"
-                  stroke-linecap="round"
-                  stroke-width="2.8"
-                  style="animation: 1s linear infinite spinner-rotation; transform-origin: 50% 50%; transform-box: fill-box;"
-                />
-              </svg>
-            </span>
+              noData
```
Comment (chetanvekariya-harness) [inline src/modules/10-common/components/UserGroupsInput/__test__/__snapshots__/UserGroupsInput.test.tsx.snap:232]: Spinner has changed to noData view. Is this intended ?

**Review by chetanvekariya-harness — APPROVED** [review]

File: src/modules/10-common/components/UserGroupsInput/__test__/__snapshots__/UserGroupsInput.test.tsx.snap (hunk 1 above)
Comment (rameshx) [reply src/modules/10-common/components/UserGroupsInput/__test__/__snapshots__/UserGroupsInput.test.tsx.snap:232]: @chetanvekariya-harness Yes, the mocked function has `loading: false` and returns no data, so this is fine.

**Review by ishantmehtaharness — APPROVED** [review]
LGTM

**Pull request merged**